azurerm/internal/services/postgres/resource_arm_postgresql_server.go

```diff
@@ -22,6 +22,14 @@ import (
 	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
 )
 
+func ValidatePSQLServerName(i interface{}, k string) (_ []string, errors []error) {
+	if m, regexErrs := validate.RegExHelper(i, k, `^[a-z][0-9a-z][-0-9a-z]{0,60}[0-9a-z]$`); !m {
```

Comment:
I think this may be too restrictive, `0-b` is a valid (albeit unlikely) name, which this regex prevents.
I think this maybe : `^[a-z0-9][-0-9a-z]{0,61}[0-9a-z]$`

Reply:
I tried it with a 0 in front and the portal forbids it =/ the api might not... but then the api allows any character and just hangs when its invalid

Reply:
Looks like the portal is wrong, API works just fine with a number! good spot :) updated!

Reply:
Looks like the portal is wrong, API works just fine with a number! good spot :) updated!